symbols/overlay29.yml

```diff
@@ -1488,6 +1522,19 @@ overlay29:
         r1: Target
         r2: If true, moves with a max Ginseng boost != 99 will be ignored
         return: True if the target has at least one super effective move against the user, false otherwise.
+    - name: TryEatItem
+      address:
+        EU: 0x22FBC20
+        NA: 0x22FB214
+      description: |-
+        The user attempts to eat an item from the target.
+        
+        The function tries to eat the target's held item first. If that's not possible and the target is part of the team, it attempts to eat a random edible item from the bag instead.
+        Fun fact: The code used to select the random bag item that will be eaten is poorly coded. As a result, there's a small chance of the first edible item in the bag being picked instead of a random one. The exact chance of this happening is (N/B)^B, where N is the amount of non-edible items in the bag and B is the total amount of items in the bag.
```

Comment:
So without looking at the code I'm guessing it just picks a random item, checks if it's edible, and rerolls if not up to _B_ times?

Reply:
Yeah, pretty much. It actually builds an array with `B` random entries from the bag (which can be repeated) and then appends the numbers 0 to `B - 1` to it. Then the game iterates said array until one of the entries in it corresponds to an edible item.
No idea what they were thinking there lol.